examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,319 @@
+		c := value.(*GFMpkg.Campaign)
+		campaignsArray = append(campaignsArray, c)
+		return false
+	})
+	return GFMpkg.CampaignsToJSONString(campaignsArray)
+}
+
+func IsUserRegistered(userAddress string) bool {
+	// get the user address
+	// userAddress := std.GetOrigCaller()
+	// verify if the user exists
+	_, ok := users.Get(userAddress)
+	return ok
+}
+
+func Withdraw(campaignId string) {
+	// get the campaign if it exists
+	camp, ok := campaigns.Get(campaignId)
+	if !ok {
+		panic("campaign not found")
+	}
+
+	campaign := camp.(*GFMpkg.Campaign)
+	// get the user address
+	callerAddress := std.GetOrigCaller()
+
+	// verify that the user is the owner of the campaign
+	if callerAddress != campaign.Owner.GetAddress() {
+		panic("you are not the owner of the campaign")
+	}
+
+	// verify that the campaign is over
+	if time.Now().Unix() < campaign.Deadline.Unix() {
+		panic("the campaign is not over yet")
+	}
+
+	// XXX TO BE DISCUSSED: the funds can be withdrawn even if the goal is not reached or not?
+	// verify that the campaign is successful
+	// if campaign.Current < campaign.Goal {
+	// 	panic("the campaign is not successful")
+	// }
+
+	// get the beneficiary address
+	beneficiaryAddress := campaign.Beneficiary.GetAddress()
+	// get the amount to withdraw
+	amount := campaignsTotal[campaignId]
+	campaignsTotal[campaignId] = 0
+	// transfer the amount to the beneficiary
+	banker.SendCoins(std.CurrentRealm().Addr(), beneficiaryAddress, std.Coins{std.NewCoin("ugnot", amount)})
+	std.Emit("Withdraw", "withdraw", "withdrawn successfully")
+}
+
+func Render(path string) string {
+	if path == "" {
+		return renderHomePage()
+	}
+
+	// split the path
+	parts := strings.Split(path, "/")
+	switch {
+	case len(parts) == 2 && parts[0] == "campaign":
+		return renderCampaign(parts[1])
+	}
+	return "unknown page"
+}
+
+func renderHomePage() string {
+	balance = banker.GetCoins(std.CurrentRealm().Addr()).AmountOf("ugnot")
+	// define empty buffer
+	var b bytes.Buffer
+
+	// users counter
+	userc := 1
+	b.WriteString("# Welcome to GnoFundMe\n\n")
+	b.WriteString(ufmt.Sprintf("Balance: %d ugnot.\n", balance))
+
+	b.WriteString("# Campaigns :\n\n")
+	campaigns.Iterate("", "", func(key string, value interface{}) bool {
+		c := value.(*GFMpkg.Campaign)
+		campaignURL := ufmt.Sprintf("%scampaign/%s", campaignPrefix, key)
+		b.WriteString(
+			ufmt.Sprintf("## [Campaign %s](%s)\n", key, campaignURL),
+		)
+		b.WriteString(
+			ufmt.Sprintf(
+				"### Description: %s\n", c.Description,
+			),
+		)
+
+		return false
+	})
+	b.WriteString("# Users:\n")
+
+	users.Iterate("", "", func(key string, value interface{}) bool {
+		u := value.(*GFMpkg.User)
+		b.WriteString(
+			ufmt.Sprintf(
+				"## User #%d: %s, Key: %s\n", userc, u.GetName(), key,
+			),
+		)
+		b.WriteString(
+			ufmt.Sprintf(
+				"- email: %s\n - address: %s\n", u.GetEmail(), u.GetAddress(),
+			),
+		)
+		userc++
+		return false
+	})
+	return b.String()
+}
+
+func renderCampaign(id string) string {
```

Comment:
You could have this function as a method on the Campaign struct. Would make more sense:

```suggestion
func (c Campaign) Render() string {...}
```